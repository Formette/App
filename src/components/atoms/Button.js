import React from "react";
import PropTypes from "prop-types";
//Styles
import styled from "styled-components";
import Color from "../../styles/Colors";
import { lighten } from "polished";

const Button = props => {
  const rest = Object.assign({}, props);
  delete rest.textColor;
  return (
    <button type={props.type} className={`${props.className}`} {...rest}>
      {props.children}
    </button>
  );
};

Button.defaultProps = {
  type: "button",
  children: "text",
  color: Color.primary,
  textColor: Color.black
};

Button.propTypes = {
  type: PropTypes.string.isRequired,
  children: PropTypes.any.isRequired,
  color: PropTypes.string.isRequired,
  textColor: PropTypes.string
};

const ButtonWithStyles = styled(Button)`
  border-radius: 0;
  background: ${props => (props.color ? props.color : Color.default)};
  color: ${(props) => (props.textColor ? props.textColor : Color.black)};
  border: 0;
  cursor: pointer;
  &:hover, &:active, &:visited {
    background: ${(props) =>
      props.color ? lighten(0.1, props.color) : lighten(0.1, Color.default)};
  }
  &:focus{
    outline: 0;
    box-shadow: 0 0 0 3px ${(props) =>
    props.color ? lighten(0.1, props.color) : lighten(0.1, Color.primary)};
  }
  span{
    margin-left: 5px;
  }
`;

export default ButtonWithStyles;
