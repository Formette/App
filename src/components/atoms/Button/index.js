import PropTypes from "prop-types";
//Styles
import styled from "styled-components";
import { lighten, darken } from "polished";

const Button = styled.button`
  background: ${props =>
    props.primary
      ? props.theme.color.secondary
      : props.warning
        ? props.theme.color.yellow
        : props.danger
          ? props.theme.color.red
          : props.theme.color.default};
  color: ${props =>
    props.primary || props.warning || props.danger
      ? props.theme.color.white
      : props.theme.color.primary};
  box-shadow: ${props => props.theme.boxShadow};
  border: 0;
  border-radius: ${props => props.theme.radius};
  cursor: pointer;
  &:hover {
    background: ${props =>
      props.primary
        ? lighten(0.1, props.theme.color.secondary)
        : lighten(0.1, props.theme.color.default)} !important;
  }
  &:active,
  &:visited {
    background: ${props =>
      props.primary
        ? darken(0.1, props.theme.color.secondary)
        : darken(0.1, props.theme.color.default)} !important;
  }
  &:focus {
    outline: 0;
    box-shadow: 0 0 0 3px
      ${props =>
        props.primary
          ? lighten(0.1, props.theme.color.secondary)
          : lighten(0.1, props.theme.color.default)};
  }
  span {
    margin-left: 5px;
  }
`;

Button.defaultProps = {
  type: "button",
  children: "text"
};

Button.propTypes = {
  type: PropTypes.string.isRequired,
  children: PropTypes.any.isRequired,
  color: PropTypes.string,
  textColor: PropTypes.string
};

export default Button;
