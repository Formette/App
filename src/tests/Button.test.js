// Button.test.js
import React from 'react';
import Button from '../components/atoms/Button';
import renderer from 'react-test-renderer';

test('Button changes the state when hovered', () => {
    const component = renderer.create(
        <Button type="button" color="#000">Hello</Button>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});