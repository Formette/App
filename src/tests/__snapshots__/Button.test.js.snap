// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  className="sc-bdVaJa hyCWPD"
  color="#000"
  type="button"
>
  Hello
</button>
`;
